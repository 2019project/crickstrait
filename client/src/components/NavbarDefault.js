import React, { Component } from 'react'
import "../styles/Navbar.css";
import { Link } from "react-router-dom";
import logo from "./fakelogo.png"
 
export default class NavbarDefault extends Component {

    //Displays option inside the hamburger button
    myhamburgfunction() {

        // gets the complete navbar element
        var x = document.getElementById("navbarID");
        // checks if navbar has the required className 
        if (x.className === "navbarHS") {
        // adds another className that displays navbar options
          x.className += " responsiveHS";
        } else {
          x.className = "navbarHS";
        }
      };

    // showActive() {
    //     document.getElementById('navbarActive').style.display = "block";
    //  }
     

    render() { 
        return (
    
                <div className="navbarHS" id="navbarID">

                    <Link to="/Headstocks">

                        <div className="navbarLogoName">
                            <span className="active">
                            <img className="navbarlogoimage" src={logo} alt="logo" width="30px" height="30px"/>
                                HEAD<b>STOCKS</b>
                            </span> 
                        </div>

                    </Link>

            
                    <div className="navbarLoginOptions">
                        <span>
                            <Link to="/Login" className="linkdecornone">
                            Login
                            </Link>
                        </span> 
                    </div>
                   
                   

                    <div className="navbarStockOptions" 
                    // onClick={this.showActive}
                    >

                        <span>
                            <Link to="/StocksHome" className="linkdecornone">
                            Stocks
                            </Link>
                            {/* <div id="navbarActive"  style={{display:"none"}} class="answer_list" >
                            <hr style= {{
                                backgroundcolor:"#39abf7",
                                width: "65px", 
                                padding: "0px",
                                margin: "0px",
                                border: "1.5px solid #39abf7"}}
                                />
                            </div> */}
                        </span> 
                    
                    </div>

                    <div className= "navbarSearchBox">
                        <span  className= "navbarSearchSpan">
                        <input 
                        type="text" 
                        placeholder="Search for a company or ticker"
                         />
                        </span>
                   </div>

                    <span className="icon" onClick={this.myhamburgfunction}>
                        <i className="fa fa-bars"></i>
                    </span>
                
                   
                    

{/* <input type="button" name="answer" value="Show Div" onClick={this.showActive} /> */}
                </div>

            
        )
    }
}
