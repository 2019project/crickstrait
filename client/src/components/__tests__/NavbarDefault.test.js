import React from "react";
import { NavbarDefault } from "../NavbarDefault";
import { shallow, mount } from "enzyme";

// import renderer from 'react-test-renderer';
// import { Link } from 'react-router-dom';
// import { StaticRouter } from 'react-router'

const navbar = jest.fn();
const searchContent = jest.fn();


const wrapper = shallow(
  <NavbarDefault
    navbar={navbar}
    searchContent={searchContent}
  />
);

describe("Testing Navbar Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a division with name navbarID", () => {
    expect(wrapper.find("#navbarID")).toBeTruthy();
  });
  it("should have a division with name navbarSearchBoxDiv", () => {
    expect(wrapper.find("#navbarSearchBoxDiv")).toBeTruthy();
  });
  it("should display the market cap", () => {
    expect(wrapper.find("#navbarLogoText1").text()).toBe("HEADSTOCKS");
  });
  it("should display the Sector", () => {
    expect(wrapper.find("#navbarLogoText2").text()).toBe("STOCKS");
  });
  it("should display the industry", () => {
    expect(wrapper.find("#navbarLoginText").text()).toBe("Login");
  });
  it("should display the employee", () => {
    expect(wrapper.find("#navbarStocksText").text()).toBe("Stocks");
  });

  it("should simulate button click on Login button tab ", () => {
    wrapper.find("#navbarLoginText").simulate("click");
    expect(wrapper.state().home).toBe(false);
    expect(wrapper.state().stocks).toBe(false);
    expect(wrapper.state().login).toBe(true);
  });
  it("should simulate button click on Stocks button tab ", () => {
    wrapper.find("#navbarStocksText").simulate("click");
    expect(wrapper.state().home).toBe(false);
    expect(wrapper.state().stocks).toBe(true);
    expect(wrapper.state().login).toBe(false);
  });
  it("should simulate button click on Home button tab ", () => {
    wrapper.find("#navbarLogoText1").simulate("click");
    expect(wrapper.state().home).toBe(true);
    expect(wrapper.state().stocks).toBe(false);
    expect(wrapper.state().login).toBe(false);
  });
  
  it("checks for on Search function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#navbarSearchResultsSearchButton").simulate("click", e);
    expect(e.preventDefault).toBeCalled();
  });
  it("checks for Search button click ", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#navbarSearchResultsSearchButton").simulate("click", e);
  });
  it("should check for presense of onSearch function", () => {
    let onSearch = jest.spyOn(wrapper.instance(), "onSearch");
    expect(onSearch).toBeTruthy();
  });
  it("should check for presense of my hamburg function function", () => {
    let myhamburgfunction = jest.spyOn(wrapper.instance(), "myhamburgfunction");
    expect(myhamburgfunction).toBeTruthy();
  });

  it("should check for presense of OnChange function", () => {
    let OnChange = jest.spyOn(wrapper.instance(), "OnChange");
    expect(OnChange).toBeTruthy();
  });
  it("should check for Search input ", () => {
    expect(
      wrapper
        .find("#navbarSearchInput")
        .at(0)
        .props().placeholder
    ).toBe("Search for ticker");
  });


  // it("should simulate button click on Cancel button click ", () => {
  //   wrapper.find("#navbarSearchCancelClick").simulate("click");
  //   expect(wrapper.state().searchInputChanged).toBe(true);
  // });


  // Does Not Affect the Coverage
  // id="linknavbar1"
  // test('Link matches snapshot', () => {
  //   const component = renderer.create(
  //     <StaticRouter location="#linknavbar1" navbar={navbar}>
  //       <Link to="/companydetail/9" />
  //     </StaticRouter>
  //   );
  //   let tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // })


});


