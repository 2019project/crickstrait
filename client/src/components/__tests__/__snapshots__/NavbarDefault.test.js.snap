// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component should mount the component 1`] = `
<div
  className="navbarHS"
  id="navbarID"
>
  <Link
    to="/Headstocks"
  >
    <div
      className="navbarLogoName"
    >
      <span
        className="active"
      >
        <img
          alt="logo"
          className="navbarlogoimage"
          height="30px"
          src="fakelogo.png"
          width="30px"
        />
        HEAD
        <b>
          STOCKS
        </b>
      </span>
    </div>
  </Link>
  <div
    className="navbarLoginOptions"
  >
    <span>
      <Link
        className="linkdecornone"
        to="/Login"
      >
        Login
      </Link>
    </span>
  </div>
  <div
    className="navbarStockOptions"
  >
    <span>
      <Link
        className="linkdecornone"
        to="/StocksHome"
      >
        Stocks
      </Link>
    </span>
  </div>
  <div
    className="navbarSearchBox"
  >
    <span
      className="navbarSearchSpan"
    >
      <input
        placeholder="Search for a company or ticker"
        type="text"
      />
    </span>
  </div>
  <span
    className="icon"
    onClick={[Function]}
  >
    <i
      className="fa fa-bars"
    />
  </span>
</div>
`;
